File: C++/insertion_sort.cpp
#include <iostream>
#include <vector>

// Function for sorting a vector using insertion sort
std::vector<int> insertionSort(std::vector<int>& v) {
    for (int i = 1; i < v.size(); i++) {
        int j = i;
        while (j > 0 && v[j] < v[j - 1]) {
            std::swap(v[j], v[j - 1]);
            j--;
        }
    }
    return v;
}

int main() {
    std::vector<int> v = { 5, 4, 3, 2, 1 };
    v = insertionSort(v);

    // Print the vector
    for (int i = 0; i < v.size(); i++) {
        std::cout << v[i] << ", ";
    }

    return 0;
}

File: Python/insertion_sort.py
# Program to implement insertion sort algorithm

def insertion_sort(arr):
    for i in range(1, len(arr)):
        j = i
        while j > 0 and arr[j] < arr[j-1]:
            arr[j], arr[j-1] = arr[j-1], arr[j]
            j -= 1
    return arr

if __name__ == '__main__':
    arr = [1, 5, 2, 4, 3]
    print(insertion_sort(arr))
